File: Frontend/src/app/features/auth/models/appUser.ts
export type UserRole = 'ADMIN' | 'USER';

export interface AuthUser {
  userId: string;
  token: string;
  refreshToken: string;
  expiresAtUtc: string;
}

export interface AppUser {
  id: string;
  username?: string | null;
  role?: UserRole | null;
}

export interface AuthRequestPayload {
  username: string;
  password: string;
}

export interface SignupRequestPayload {
  firstName: string;
  lastName: string;
  email: string;
  username: string;
  password: string;
  role: UserRole;
}

File: Backend/WorldCupPool.Domain/Entities/AppUser/AppUser.cs
using WorldCupPool.Domain.Common;
using WorldCupPool.Domain.Enums;

namespace WorldCupPool.Domain.Entities
{
    public sealed class AppUser : Entity<Guid>
    {
        public string FirstName { get; private set; } = string.Empty;
        public string LastName { get; private set; } = string.Empty;
        public string Email { get; private set; } = string.Empty;
        public string UserName { get; private set; } = string.Empty;
        public string PasswordHash { get; private set; } = string.Empty;
        public string? RefreshToken { get; private set; }
        public UserRole Role { get; private set; } = UserRole.User;
        public DateTime CreatedAtUtc { get; private set; } = DateTime.UtcNow;
        public ICollection<Prediction> Predictions { get; private set; } = new List<Prediction>();


        public AppUser() { }

        public AppUser(string firstName, string lastName, string email, string username, string passwordHash, UserRole role)
        {
            Id = Guid.NewGuid();
            FirstName = firstName;
            LastName = lastName;
            Email = email;
            UserName = username;
            PasswordHash = passwordHash;
            Role = role;
            CreatedAtUtc = DateTime.UtcNow;
        }


        public void SetRefreshToken(string refreshToken)
        {
            RefreshToken = refreshToken;
        }
    }
}
